test/protocol_test.py

```diff
@@ -1274,18 +1274,30 @@ def test_one_source(self, dummy_protocol):
         p = dummy_protocol
         echo = p.ref("echo", None, "384-echo", discard=True)
         dest = p.ref("dest", None, "384-flat", discard=True)
+
```

Comment:
You're on the right track, but I think what would make this test stronger is if you used similar params in your test otherwise you may not be catching the edge cases and only finding the cases that make you test pass. Maybe try something like this out

```
    @pytest.mark.parametrize("source_volume", ['2:microliter', '60:microliter'])
    def test_one_source(self, dummy_protocol, source_volume):
        p = dummy_protocol
        echo = p.ref("echo", None, "384-echo", discard=True)
        dest = p.ref("dest", None, "384-flat", discard=True)

        one_source = echo.wells(0, 1, 2, 3).set_volume(source_volume)
        dests = dest.wells(0, 60)
        acoustic_transfer_vol = Unit('2:microliter')

        if sum([w.available_volume() for w in one_source]) < (len(dests)*acoustic_transfer_vol):
            with pytest.raises(RuntimeError):
                p.acoustic_transfer(
                    one_source,
                    dests,
                    acoustic_transfer_vol,
                    one_source=True
                )
        else:
            p.acoustic_transfer(
                one_source,
                dests,
                acoustic_transfer_vol,
                one_source=True
            )
            for w in one_source:
                assert w.volume >= w.container.container_type.dead_volume_ul
```

Reply:
I definitely think this is a better way to structure the test so ill be pushing with that change, but im not sure what you mean by having "similar" params. which params should be similar to what?

Reply:
The parameters to the instruction should be similar the only difference should be the volume in the source wells. In your OG test you were using different combination of wells for each case (ie: RuntimeError and no error). The test should be focusing on whether the change in source volume you made will raise an error when there isn't enough source volume.

Reply:
ok yeah, isolate one thing at a time to test. in the new commits i just threw in a couple extra things to test via parameterize for good measure